src/domain_bridge/domain_bridge.cpp

```diff
@@ -453,6 +459,30 @@ class DomainBridgeImpl
       default:
         unreachable();
     }
+    if (
+      domain_bridge::TopicBridgeOptions::AutoRemove::OnNoPublisher ==
+      topic_options.auto_remove() ||
+      domain_bridge::TopicBridgeOptions::AutoRemove::OnNoPublisherOrSubscription ==
+      topic_options.auto_remove())
+    {
+      this->wait_for_graph_events_.register_on_no_publisher_callback(
+        topic, from_domain_node, [this, topic_bridge]() {
+          this->bridged_topics_[topic_bridge] = {nullptr, nullptr};
```

Comment:
It looks like there's a data race here (and inside `create_bridge` above, in retrospect). Should we put a lock guard on `bridged_topics_`?

Reply:
Yes, there's a data race here.
I think that adding a mutex is reasonable, I'll take a look.

Reply:
fixed in https://github.com/ros2/domain_bridge/pull/63/commits/0fbe05b99715709866ecaadcbb23caf52ca9cfcb